### ISSUE TITLE
1.12-RC1: Debian build randomly hangs (during call to "/liferea --introspect-dump=function.txt,dump.xml")

### ISSUE THREAD

**Issue by paulgevers** [lwindolf/liferea#459]
Hi,

I am rather reluctant to report this issue here, because I am very unsure if the issue lies with liferea, or with g-ir-scanner (or even with python subprocess). In Debian I have a severe bug reported against liferea because the build *sometimes* hangs during introspection¹.

I have already played around with in the g-ir-scanner code to add a timeout and to add file flushing of the input functions.txt file (that last item seems stupid anyways, because the dump.xml file is always available after the call to liferea). So it seems that indeed liferea, never returns. Except that when I run the call to liferea directly in a while loop, it never (so far) showed the issue. I am really at a loss, but if I don't have a solution, it may very well be that the next Debian release ships without liferea.

The call does produce some scary output though (which often doesn't hamper progress):
```
(liferea:28710): GLib-GIO-CRITICAL **: g_settings_get_value: assertion 'G_IS_SETTINGS (settings)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_get_string: assertion 'value != NULL' failed
(liferea:28710): GLib-CRITICAL **: g_variant_unref: assertion 'value != NULL' failed
(liferea:28710): GLib-GIO-CRITICAL **: g_settings_get_value: assertion 'G_IS_SETTINGS (settings)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_get_type: assertion 'value != NULL' failed
(liferea:28710): GLib-CRITICAL **: g_variant_type_is_subtype_of: assertion 'g_variant_type_check (type)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_get_boolean: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_BOOLEAN)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_unref: assertion 'value != NULL' failed
(liferea:28710): GLib-GIO-CRITICAL **: g_settings_get_value: assertion 'G_IS_SETTINGS (settings)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_get_type: assertion 'value != NULL' failed
(liferea:28710): GLib-CRITICAL **: g_variant_type_is_subtype_of: assertion 'g_variant_type_check (type)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_get_boolean: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_BOOLEAN)' failed
(liferea:28710): GLib-CRITICAL **: g_variant_unref: assertion 'value != NULL' failed
(liferea:28710): GLib-GObject-WARNING **: invalid (NULL) pointer instance
(liferea:28710): GLib-GObject-CRITICAL **: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
(liferea:28710): GLib-GObject-WARNING **: invalid (NULL) pointer instance
(liferea:28710): GLib-GObject-CRITICAL **: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
```

¹ https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=846144

**Comment by Leiaz** [comment]
I've removed the Gtk Doc warnings. I haven't tried to reproduce the random hang, but I'm still getting the assert errors.

They look like Liferea's `conf_` functions are being called without calling `conf_init` first ...

**Comment by lwindolf** [comment]
@paulgevers Can you retest using the merged fixes in #460 ?

**Comment by paulgevers** [comment]
@lwindolf Using the fixes from #460 doesn't solve the issue.

The good news for liferea in Debian is that the bug is no longer a liferea specific bug, but rather a bug against g-ir-scanner¹. (The liferea bug got closed by one of the gobject-introspection maintainers and release managers). Feel free to not spend too much time on this issue if you can use the time better elsewhere.

 ¹ https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=846782

**Comment by lwindolf** [comment]
Closing this as I see no way to fix it.
